stuffer/s2n_stuffer.c

```diff
@@ -72,17 +71,10 @@ int s2n_stuffer_growable_alloc(struct s2n_stuffer *stuffer, const uint32_t size)
 
 int s2n_stuffer_free(struct s2n_stuffer *stuffer)
 {
-    if (stuffer->alloced == 0) {
-        return 0;
-    }
-    if (stuffer->wiped == 0) {
-        GUARD(s2n_stuffer_wipe(stuffer));
+    if (stuffer->alloced) {
+        GUARD(s2n_free(&stuffer->blob));
     }
-
-    GUARD(s2n_free(&stuffer->blob));
-
-    stuffer->blob.data = NULL;
-    stuffer->blob.size = 0;
+    *stuffer = (struct s2n_stuffer) {0};
```

Comment:
is this any cheaper than an `memset`?

Reply:
They're going to be basically equal cost. `{0}` has the nice property that it does the right thing on cases where the default value is not the same as bitwise `0`: e.g. pointers on weird architectures which use non-zero bit patterns for null http://c-faq.com/null/machnon0.html
And personally, I find it easier to read.